lib/crawly/manager.ex

```diff
@@ -85,7 +94,7 @@ defmodule Crawly.Manager do
     delta = items_count - state.prev_scraped_cnt
     Logger.info("Current crawl speed is: #{delta} items/min")
 
-    case Application.get_env(:crawly, :closespider_itemcount, :disabled) do
+    case Utils.get_settings(:closespider_itemcount, state.name, :disabled) do
       :disabled ->
         :ignored
```

Comment:
I am not sure. The second branch of the case will not catch the atom :(.

For example:
```
iex(1)> 2 < :test
true
```

Reply:
How about this:
```elixir
def handle_info(:operations, state) do
    ....
    limit = Utils.get_settings(:closespider_itemcount, state.name)
    check_itemcount(state.name, cnt, limit)
    ....
end

defp check_itemcount(spider_name, current, limit) when current > limit and is_integer(limit) do
   # close spider
end
defp check_itemcount(_, _, _), do: :ok
```
much cleaner and readable.

With this, we also don't require a `:disabled` atom to be passed. If nil is returned, it will just continue.

Reply:
done